python/requirements.txt

```
@@ -85,7 +85,7 @@ pytest-lazy-fixture
 pytest-timeout
 pytest-virtualenv
 scikit-learn==0.22.2
-tensorflow==2.5.1
+tensorflow
```

let's pin it to the same version as in _dl, i.e. `2.6.2` - these are specifically test requirements so we probably want to pin the same version.

---

```suggestion
tensorflow~=2.6.2  # sync with tensorflow-probability version in python/requirements/ml/requirements_dl.txt
```
Something like this?

---

I'm thinking we can actually just remove `tensorflow` from `requirements.txt` and just keep it only in `requirements_dl.txt`.

---

Also I would say exact pin is probably safer- we do have Dependabot setup to automatically make PRs to upgrade these dependencies over time.